components/style/values/specified/basic_shape.rs

```diff
@@ -756,21 +756,21 @@ impl Parse for BorderRadius {
 
 fn parse_one_set_of_border_values(context: &ParserContext, mut input: &mut Parser)
                                  -> Result<[LengthOrPercentage; 4], ()> {
-    let a = try!(LengthOrPercentage::parse(context, input));
+    let a = try!(LengthOrPercentage::parse_non_negavite(input));
```

Comment:
right, that was bad.

I'm just curious. Would this have worked ? Also parse takes context and input, while parse_non_negative takes only input. Why is that?